spec/requests/db_capacity_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+describe '[DB Capacity]', type: :request do
+  it 'check the amount of DynamoDB read/write capacity used in performing basic tasks' do
+    Dynamoid.adapter.monitor("Log in") do
+      login
+    end
+
+    Dynamoid.adapter.monitor("Create an incident w/ 1 civilian and 1 officer") do
+      create_complete_incident
+    end
+
+    Dynamoid.adapter.monitor("Review an incident") do
+      visit_status :in_review
+      click_link "View"
+    end
+
+    Dynamoid.adapter.monitor("Create an incident w/ 3 civilians and 3 officers") do
+      create_complete_incident 3, 3
+    end
+  end
+end
+
+module Dynamoid
+  class Adapter
+    attr_reader :read_capacity_used, :write_capacity_used
+
+    # Given a block monitor the read and write capacity taken
+    # to execute the task.
+    def monitor(description)
+      puts ""
+      puts "(#{description})"
+
+      @read_capacity_used = Hash.new { |h, k| h[k] = 0 }
+      @write_capacity_used = Hash.new { |h, k| h[k] = 0 }
+
+      @monitoring = true
+      yield
+      @monitoring = false
+
+      puts "  Read capacity used: #{Dynamoid.adapter.read_capacity_used.values.reduce(:+)}"
+      Dynamoid.adapter.read_capacity_used.each do |table, capacity|
+        puts "    #{table}: #{capacity}"
+      end
+
+      puts "  Write capacity used: #{Dynamoid.adapter.write_capacity_used.values.reduce(:+)}"
+      Dynamoid.adapter.write_capacity_used.each do |table, capacity|
+        puts "    #{table}: #{capacity}"
+      end
+    end
+
+    # DynamoDB's built-in `Adapter#benchmark` method simply times operations.
+    # Instead, let's benchmark read/write capacity usage.
+    def benchmark(method, *args)
+      result = yield
+
+      if @monitoring && args
+        case method
+        when 'get_item'
+          @read_capacity_used[args[0][0]] += 1
```

Comment:
So we're assuming each transaction is under 4k? Which is fine for simplicity, but let's change the output text to be accurate. Instead of "Read capacity used" you should say "Read operations" etc. And maybe also print somewhere that we are only counting >4k reads as a single one.